c/driver_manager/adbc_driver_manager.cc

```diff
@@ -114,7 +130,186 @@ struct OwnedError {
   }
 };
 
+#ifdef _WIN32
+using char_type = wchar_t;
+
+std::string Utf8Encode(const std::wstring& wstr) {
+  if (wstr.empty()) return std::string();
+  int size_needed = WideCharToMultiByte(
+      CP_UTF8, 0, &wstr[0], static_cast<int>(wstr.size()), NULL, 0, NULL, NULL);
+  std::string str_to(size_needed, 0);
+  WideCharToMultiByte(CP_UTF8, 0, &wstr[0], static_cast<int>(wstr.size()), &str_to[0],
+                      size_needed, NULL, NULL);
+  return str_to;
+}
+
+std::wstring Utf8Decode(const std::string& str) {
+  if (str.empty()) return std::wstring();
+  int size_needed =
+      MultiByteToWideChar(CP_UTF8, 0, &str[0], static_cast<int>(str.size()), NULL, 0);
+  std::wstring wstr_to(size_needed, 0);
+  MultiByteToWideChar(CP_UTF8, 0, &str[0], static_cast<int>(str.size()), &wstr_to[0],
+                      size_needed);
+  return wstr_to;
+}
+
+#else
+using char_type = char;
+#endif
+
 // Driver state
+struct DriverInfo {
+  std::string manifest_file;
+  std::string driver_name;
+  std::filesystem::path lib_path;
+  std::string entrypoint;
+
+  std::string version;
+  std::string source;
+};
+
+#ifdef _WIN32
+class RegistryKey {
+ public:
+  RegistryKey(HKEY root, const std::wstring_view subkey) noexcept
+      : root_(root), key_(nullptr) {
+    status_ = RegOpenKeyExW(root_, subkey.data(), 0, KEY_READ, &key_);
+  }
+
+  ~RegistryKey() {
+    if (is_open() && key_ != nullptr) {
+      RegCloseKey(key_);
+      key_ = nullptr;
+      status_ = ERROR_REGISTRY_IO_FAILED;
+    }
+  }
+
+  HKEY key() const { return key_; }
+  bool is_open() const { return status_ == ERROR_SUCCESS; }
+  LSTATUS status() const { return status_; }
+
+  std::wstring GetString(const std::wstring& name, std::wstring default_value) {
+    if (!is_open()) return default_value;
+
+    DWORD type = REG_SZ;
+    DWORD size = 0;
+    auto result = RegQueryValueExW(key_, name.c_str(), nullptr, &type, nullptr, &size);
```

Comment:
We can use `data()` with C++11 or later because `data()` with C++11 or later returns `NULL` terminated C string:

```suggestion
    auto result = RegQueryValueExW(key_, name.data(), nullptr, &type, nullptr, &size);
```